File: lib/entity/message_entity.dart
import 'dart:convert';

import 'node_entity.dart';

/// 消息实体
class MessageEntity {
  // 随机码
  int rand;

  // 自定义整数
  int customInt;

  // 自定义值
  String customStr;

  // 是否已读
  bool read;

  // 节点列表
  List<NodeEntity> elemList;

  // 当前登录用户是否是发送方
  bool self;

  // 消息id
  String id;

  // 消息唯一id
  int uniqueId;

  // 序列号
  int seq;

  // 时间戳
  int timestamp;

  // 对方是否已读
  bool peerReaded;

  MessageEntity({
    this.rand,
    this.customInt,
    this.read,
    this.elemList,
    this.self,
    this.id,
    this.uniqueId,
    this.seq,
    this.customStr,
    this.timestamp,
    this.peerReaded,
  });

  MessageEntity.fromJson(Map<String, dynamic> json) {
    rand = json['rand'];
    customInt = json['customInt'];
    read = json['read'];
    if (json['elemList'] != null) {
      List<NodeEntity> entitys = [];
      for (var item in json['elemList']) {
        entitys.add(NodeEntity.getEntity(item));
      }
      elemList = entitys;
    }
    self = json['self'];
    id = json['id'];
    uniqueId = json['uniqueId'];
    seq = json['seq'];
    customStr = json['customStr'];
    timestamp = json['timestamp'];
    peerReaded = json['peerReaded'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['rand'] = this.rand;
    data['customInt'] = this.customInt;
    data['read'] = this.read;
    if (this.elemList != null) {
      data['elemList'] = elemList == null ? null : jsonEncode(elemList);
    }
    data['self'] = this.self;
    data['id'] = this.id;
    data['uniqueId'] = this.uniqueId;
    data['seq'] = this.seq;
    data['customStr'] = this.customStr;
    data['timestamp'] = this.timestamp;
    data['peerReaded'] = this.peerReaded;
    return data;
  }
}

File: ios/Classes/entity/MessageEntity.swift
import ImSDK

//
//  MessageEntity.swift
//  tencent_im_plugin
//
//  Created by 蒋具宏 on 2020/2/10.
//
public class MessageEntity : NSObject{
    /**
     * 消息ID
     */
    var id : String?;
    
    /**
     * 唯一ID
     */
    var uniqueId : UInt64?;
    
    /**
     * 对方是否已读
     */
    var peerReaded : Bool?;
    
    /**
     * 自己是否已读
     */
    var read : Bool?;
    
    /**
     * 当前登录用户是否是发送方
     */
    var `self` : Bool?;
    
    /**
     * 自定义整数
     */
    var customInt : Int32?;
    
    /**
     * 自定义值
     */
    var customStr : String?;
    
    /**
     * 消息时间戳
     */
    var timestamp : Date?;
    
    /**
     * 消息发送方
     */
    var sender : String?;
    
    /**
     * 会话ID
     */
    var sessionId : String?
    
    /**
     * 发送人->用户信息
     */
    private var userInfo : TIMUserProfile?;
    
    /**
     * 发送人->群成员信息
     */
    private var groupMemberInfo : TIMGroupMemberInfo?;
    
    /**
     * 节点内容
     */
    private var elemList : [TIMElem]?;
    
    /**
     * 消息状态
     */
    private var status : TIMMessageStatus?;
    
    /**
     * 会话实体
     */
    private var sessionEntity : SessionEntity?;
    
    override init() {
    }
    
    init(message : TIMMessage) {
        super.init();
        self.id = message.msgId();
        self.uniqueId = message.uniqueId();
        self.peerReaded = message.isPeerReaded();
        self.read = message.isReaded();
        self.`self` = message.isSelf();
        self.customInt = message.customInt();
        self.customStr = String(data: message.customData()!, encoding: String.Encoding.utf8);
        self.timestamp = message.timestamp();
        self.elemList = TencentImUtils.getArrrElement(message: message);
        self.groupMemberInfo = message.getSenderGroupMemberProfile();
        self.sender = message.sender();
        self.sessionId = message.getConversation().getReceiver();
        self.status = message.status();
    }
}